lib/Alchemy/Phrasea/Controller/Prod/ExportController.php

```diff
@@ -191,40 +192,68 @@ public function exportMail(Request $request)
         $token = $this->getTokenManipulator()->createEmailExportToken(serialize($list));
 
         if (count($destMails) > 0) {
-            //zip documents
-            \set_export::build_zip(
-                $this->app,
-                $token,
-                $list,
-                $this->app['tmp.download.path'].'/'. $token->getValue() . '.zip'
-            );
 
-            $remaingEmails = $destMails;
+            //  if the service worker plugin is installed and enabled , use it
+            if ($this->app['plugins.manager']->isEnabled('phraseanet-plugin-services')) {
```

Comment:
no. the core code must not contain "if(x) then event() else hardcode()".
The app should not test if a plugin is installed (app must not know plugins).
The job is event-driven or hardcoded, not both.
As event-driven, the controller simply dispatch a core (internal) event - that's all -, and a core event-listener does the hardcoded job.
When the "plugin-service" is installed, it overrides the core listener and does his own way.